### PR TITLE
Faster turning for pacman and refactor for Ghost animation

### LINKED ISSUE #122: Longer Key press slows rendering of pacman
renderStopOnKeypress

The `setState()` on arrow key press, triggers `setState()` one after another leading to the pacman rendering just stopping.

### LINKED ISSUE #123: Refactor PacmanEntity
Currently, `PacmanEntity` is the only animating entity. Ghost also need to be animated. Refactor the `PacmanEntity` into the `AnimateEntity` so that both `Ghost` and `PacmanEntity`can use it without repeating code.

### PR DESCRIPTION
**Pull request by shivangg**
For the animation of the ghost, the previous location of ghosts will need to be stored. Current PR facilitates that future work of Ghost Animations.

Resolves #123 , Resolves #122

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,6 +43,6 @@ Build the Multiplayer PacMan game which would work smoothly with various fun fea
 
 ### URLS: 
 
-- Game-staging: https://5c89a982fb0eb10008c6c763--youthful-dubinsky-b46f9a.netlify.com/
+- Game-staging: https://youthful-dubinsky-b46f9a.netlify.com/
```
Comment (kaushikb11) [inline README.md:46]: Finally. It was so embarrassing.

Comment (shivangg) [reply README.md:46]: :fire: :rocket:

Comment (nimish-gupta) [reply README.md:46]: You can create a alias for this, so that it can be remain fixed

**Review by nimish-gupta — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (shivangg) [reply README.md:46]: Previously, we had appended the commit to the URL which linked to the deploy of **that commit**. 

Now, we have removed the commit from URL. By default, the `master` is deployed to this URL. So, the link will stay fixed.

File: client/src/components/PacmanGame/AnimateEntity.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+class AnimateEntity extends Component {
+  componentDidMount() {
+    this.animate(this.node, this.center);
+  }
+
+  componentDidUpdate() {
+    this.animate(this.node, this.center);
+  }
+
+  killerCell = ({
```
Comment (kaushikb11) [inline client/src/components/PacmanGame/AnimateEntity.jsx:17]: What is killerCell?

File: client/src/components/PacmanGame/AnimateEntity.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    entity, gridSize, ghostIndex, gridX, gridY,
+  }) => {
+    const image = new window.Image(gridSize, gridSize);
+    image.src = (entity === 'scatterGhost') ? ghostImages[4] : ghostImages[ghostIndex];
+    return (
+      <Image
+        ref={(node) => {
```
Comment (kaushikb11) [inline client/src/components/PacmanGame/AnimateEntity.jsx:24]: The use of ref is necessary?

File: client/src/components/PacmanGame/AnimateEntity.jsx (hunk 2 above)
Comment (shivangg) [reply client/src/components/PacmanGame/AnimateEntity.jsx:17]: Ghosts and ScatterGhosts.

File: client/src/components/PacmanGame/AnimateEntity.jsx (hunk 3 above)
Comment (shivangg) [reply client/src/components/PacmanGame/AnimateEntity.jsx:24]: Absolutely. No way to do it any other way. 
By `react-konva` creator: [Getting reference to Konva objects](https://github.com/konvajs/react-konva#getting-reference-to-konva-objects)

**Review by kaushikb11 — APPROVED** [review]

**Pull request merged**